### PR TITLE
Pipeline 292 alex userassertions api

### PR DESCRIPTION
**Pull request by alexhuang091**
- fixed apis to get/add/delete user assertions.
- Also changed `StoreDAO.get()` to return an `Optional` so anyone uses it should check if the return value is `null`

### CODE REVIEW

**Comment by djtfmartin** [comment]
@alexhuang091 tests are still failing.

I suggest we convert this to draft until tests are fixed.

**Review by djtfmartin — CHANGES_REQUESTED** [review]
We need to fix tests

**Comment by alexhuang091** [comment]
> We need to fix tests

Hi Dave, the failures are in test cases for other code, not for assertion related code. I'll try to fix them.

File: src/main/java/au/org/ala/biocache/dto/QualityAssertion.java [hunk 1]
```diff
@@ -25,20 +27,27 @@
...
     public QualityAssertion() {
+        uuid = UUID.randomUUID().toString();
+        // to ISO date format to compatible with existing database records
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        created = simpleDateFormat.format(new Date());
     }
 
     public QualityAssertion(ErrorCode errorCode, String comment) {
+        super();
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/dto/QualityAssertion.java:43]: Why the call to super() ?

File: src/main/java/au/org/ala/biocache/dto/QualityAssertion.java [hunk 2]
```diff
@@ -25,20 +27,27 @@
...
         this.name = errorCode.name;
         this.code = errorCode.code;
         this.comment = comment;
     }
 
     public QualityAssertion(ErrorCode errorCode, Integer qaStatus) {
+        super();
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/dto/QualityAssertion.java:50]: Why the call to super() ?

File: src/main/java/au/org/ala/biocache/service/AssertionService.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            if (!combinedAssertions.isEmpty()) {
+                store.put(recordUuid, combinedAssertions);
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    // recordId + userId + code should be unique for normal assertions
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/service/AssertionService.java:129]: Probably worth javadoc-ing this method, a description at the top

File: src/main/java/au/org/ala/biocache/util/OccurrenceUtils.java [hunk 4]
```diff
@@ -38,16 +42,18 @@
...
+        try {
+            list = searchDAO.findByFulltext(idRequest);
+        } catch (Exception ignored) {
+            logger.debug("Failed to find occurrence with id " + uuid);
         }
-        return null;
+        return (list != null && list.size() > 0) ? list.get(0) : null;
     }
 
 //    private FullRecord[] getFullRecord(String uuid, SolrDocumentList result) {
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/util/OccurrenceUtils.java:59]: Remove commented code

File: src/main/java/au/org/ala/biocache/util/OccurrenceUtils.java [hunk 5]
```diff
@@ -86,8 +92,7 @@ private SolrDocument lookupRecordFromSolr(String uuid) throws Exception {
 //    }
 
 
-    public SolrDocument getOcc(String uuid) throws Exception {
-        SolrDocument sd = lookupRecordFromSolr(uuid);
+    public SolrDocument getOcc(String uuid) {
 
 //        Map<String, List<QualityAssertion>> assertions = new HashMap<String, List<QualityAssertion>>();
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/util/OccurrenceUtils.java:97]: remove commented code

File: src/main/java/au/org/ala/biocache/web/OccurrenceController.java [hunk 6]
```diff
@@ -1511,8 +1511,8 @@ private Object getOccurrenceInformation(String uuid, String im, HttpServletReque
                         ua.setUserId(authService.getDisplayNameFor(userId));
                     }
                 }
-                sd.addField("dbUserAssertions", userAssertions);
-            }
+                sd.addField("dbUserAssertions", assertions);
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/web/OccurrenceController.java:1514]: Is this `dbUserAssertions` field used ? Its not referenced elsewhere so I don't think it appears in any output

**Review by djtfmartin — APPROVED** [review]
thanks @alexhuang091 

Just minor things. Feel free to fix up and just merge without another review.

**Pull request merged**